spyder/plugins/editor/widgets/codeeditor.py

```diff
@@ -1582,19 +1582,29 @@ def paste(self):
         text has 'CRLF' EOL chars
         """
         clipboard = QApplication.clipboard()
-        # This is here to prevent pasting mime data in the Editor.
-        # See issue 8566 for the details.
+        text = to_text_string(clipboard.text())
+        # Have copied file and folder urls pasted as text paths.
+        # See issue: 8566 and PR: 8576 for details.
+        urls = []
         if clipboard.mimeData().hasUrls():
-            return
-        else:
-            text = to_text_string(clipboard.text())
-            if len(text.splitlines()) > 1:
-                eol_chars = self.get_line_separator()
-                text = eol_chars.join((text + eol_chars).splitlines())
-                clipboard.setText(text)
-            # Standard paste
-            TextEditBaseWidget.paste(self)
-            self.document_did_change(text)
+            urls = clipboard.mimeData().urls()
+            if len(urls) > 1:
+                text = ''.join('"' + url.toLocalFile().replace(osp.os.sep, '/')
+                               + '",' + '\n' for url in urls)[:-2]
+            else:
+                text = urls[0].toLocalFile()
+        if len(text.splitlines()) > 1:
+            eol_chars = self.get_line_separator()

```

Comment:
Move this to the top of the function, right after the ``clipboard = QApplication.clipboard()`` call per @bcolsen 's previous review.